### PR TITLE
Build ChainerX documentation on ReadTheDocs

### PR DESCRIPTION
**Pull request by kmaehashi**
See https://docs.readthedocs.io/en/latest/yaml-config.html for the reference of RTD YAML file.

### CODE REVIEW

**Comment by kmaehashi** [comment]
I confirmed the docs can build successfully.

File: docs/environment.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Conda environment definition to build documentation on RTD
```
Comment (niboshi) [inline docs/environment.yml]: How about something like `rtd_conda.yml` to make it more explicit about the purpose?

File: readthedocs.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 name: chainer
 type: sphinx
 base: docs/source
+conda:
+  file: docs/environment.yml
 python:
+  version: 3
```
Comment (niboshi) [inline readthedocs.yml]: How about `3.5`?

File: setup.py [hunk 3]
```diff
@@ -187,6 +187,10 @@ def find_any_distribution(pkgs):
 
 
 build_chainerx = 0 != int(os.getenv('CHAINER_BUILD_CHAINERX', '0'))
+if os.getenv('READTHEDOCS', None):
```
Comment (niboshi) [inline setup.py]: How about `== 'True'`?

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM

**Pull request merged**